adaway-linux.sh

```diff
@@ -39,53 +39,59 @@ fi
 
 # preparing temporary directory
 if [ -d "${TMPDIR}" ]; then
-    echo "[i] deleting directory ${TMPDIR}"
+    echo "[i] Deleting directory ${TMPDIR}"
     rm -r "${TMPDIR}"
 fi
 
-echo "[i] creating temporary directory ${TMPDIR}"
+echo "[i] Creating temporary directory ${TMPDIR}"
 mkdir -p "${TMPDIR}"
 
 # add domains from hosts-server listet in hostssources.lst
 while read src; do
     if [[ $src != "#*" ]] ; then
         # only insert entries redirecting to 127.0.0.1 or 0.0.0.0 (everything else might be a security nightmare)
-        curl --progress-bar -L "${src}" | sed 's/\r/\n/' | sed 's/\s\+/\t/' | sed 's/0\.0\.0\.0/127.0.0.1/' | grep '127\.0\.0\.1' >> "${TMPDIR}/hosts.downloaded"
+        curl --progress-bar -L "${src}" | sed 's/\r/\n/' | sed 's/\s\+/\t/' | sed 's/0\.0\.0\.0/127.0.0.1/' | grep '127\.0\.0\.1' >> "${TMPDIR}hosts.downloaded"
     else
         echo "[i] skipping $src"
     fi
 done <hostssources.lst
 
+echo "[i] Cleanup ${TMPDIR}hosts.downloaded and merge with original content"
+grep '^\([0-9]\|:\)' "${TMPDIR}hosts.downloaded" > "${TMPDIR}temp" && mv "${TMPDIR}temp" "${TMPDIR}hosts.downloaded" # Remove all lines that are not hosts entries
+sed 's/\(\t\| \)\+/ /g' "${TMPDIR}hosts.downloaded" > "${TMPDIR}temp" && mv "${TMPDIR}temp" "${TMPDIR}hosts.downloaded" # Replace all whitespace with spaces for neatness
+sed 's/ *\#.*/\r/g' "${TMPDIR}hosts.downloaded" > "${TMPDIR}temp" && mv "${TMPDIR}temp" "${TMPDIR}hosts.downloaded" # Remove remaining comments
+sed 's/^127\.0\.0\.1/0\.0\.0\.0/g' "${TMPDIR}hosts.downloaded" > "${TMPDIR}temp" && mv "${TMPDIR}temp" "${TMPDIR}hosts.downloaded" # Replace 127.0.0.0 with 0.0.0.0 (Shorter is faster!)
+sed '/localhost/d' "${TMPDIR}hosts.downloaded" > "${TMPDIR}temp" && mv "${TMPDIR}temp" "${TMPDIR}hosts.downloaded" # Remove additional localhost entries possibly picked up from sources
+uniq <(sort "${TMPDIR}hosts.downloaded") >> "${TMPDIR}hostsTemp"
```

Comment:
Yeah, I'm not entirely familiar with sed, but if there's a way of directly writing to a file it's certainly a better decision. Replacing 127.0.0.1 with 0.0.0.0 is just a small way of reducing the hosts size considerably. Also, I meant to comment on the way I find old localhost entries, because if a blocked domain were to contain "localhost", it would remove that from the list. Better regex is needed!